lib/matplotlib/backend_bases.py

```diff
@@ -3520,6 +3555,13 @@ def tool_trigger_event(self, name, sender=None, canvasevent=None,
             warnings.warn("%s is not a tool controlled by Navigation" % name)
             return
 
+        tool = self.get_tool(name)
+
+        rg = getattr(tool, 'radio_group', None)
+        if rg and self._radio_groups[rg]['options'].get('required') and \
```

Comment:
Perhaps I should add a comment here to explain this line, basically *if*:
1. We deal with a toggle (aka radio) tool and
2. It has required set as True or something that equates to true (get will automatically return None if not set), and
3. We try to deactivate the tool (self._toggled[rg] == name)...
4. Without activating another tool (sender is not self, dodgy in the future, but works for now, the only method that has sender=Navigation, comes from _handle_toggle to deactivate this tool before activating another).

then we don't want to reactivate this tool.

Though as I say I don't like this if statement any more, I think it too specific for this method, I would rather have the more obvious ``if not self.can_trigger(tool, sender)``.